src/libs/actions/App.js

```diff
@@ -45,29 +45,29 @@ Onyx.connect({
     },
 });
 
-let policyIDList = [];
+let allPolicies = [];
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.POLICY,
     waitForCollectionCallback: true,
     callback: (policies) => {
-        policyIDList = _.compact(_.pluck(policies, 'id'));
+        allPolicies = policies;
     },
 });
 
-// When we reconnect the app, we don't want to fetch workspaces created optimistically while offline since they don't exist yet in the back-end.
-// If we fetched them then they'd return as empty objects which would clear out the optimistic policy values initially created locally.
-// Once the re-queued call to CreateWorkspace returns, the full contents of the workspace excluded here should be correctly saved into Onyx.
-let policyIDListExcludingWorkspacesCreatedOffline = [];
-Onyx.connect({
-    key: ONYXKEYS.COLLECTION.POLICY,
-    waitForCollectionCallback: true,
-    callback: (policies) => {
-        const policiesExcludingWorkspacesCreatedOffline = _.reject(policies,
-            policy => lodashGet(policy, 'pendingAction', null) === CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD
-            && lodashGet(policy, 'type', null) === CONST.POLICY.TYPE.FREE);
-        policyIDListExcludingWorkspacesCreatedOffline = _.compact(_.pluck(policiesExcludingWorkspacesCreatedOffline, 'id'));
-    },
-});
+/**
+ * When we reconnect the app, we don't want to fetch workspaces created optimistically while offline since they don't exist yet in the back-end.
+ * If we fetched them then they'd return as empty objects which would clear out the optimistic policy values initially created locally.
+ * Once the re-queued call to CreateWorkspace returns, the full contents of the workspace excluded here should be correctly saved into Onyx.
+ *
+ * @param {Array} policies
+ * @return {Array<String>} array of policy ids
+*/
+function getPolicyIDListExcludingWorkspacesCreatedOffline(policies) {
+    const policiesExcludingWorkspacesCreatedOffline = _.reject(policies,
+        policy => lodashGet(policy, 'pendingAction', null) === CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD
+        && lodashGet(policy, 'type', null) === CONST.POLICY.TYPE.FREE);
```

Comment:
Why does it matter if they are free or not? The type of the policy has nothing to do with if the policy was created while it was offline or not (taking the name of the function at face value). I suggest removing this piece of logic.

Reply:
I do not know much about the policy types but I agree with removing the free policy check,

Reply:
Agreed, looks like an unnecessary check

Reply:
Updated, also refactored the code a bit into something that looks more readable to me.